linkerd/app/outbound/src/http/logical/policy/route/metrics/labels.rs

```diff
@@ -215,37 +211,6 @@ impl EncodeLabelSet for GrpcRsp {
     }
 }
 
-// === impl MatchedRoute ===
-
-impl<T, M, F, P> super::super::MatchedRoute<T, M, F, P> {
-    /// Returns a [`RouteLabelExtract`].
-    ///
-    /// The extractor returned by this function provides a [`ExtractParam<P, T>`] implementation
-    /// that can be used to acquire the route-level labels corresponding to a given outbound
-    /// request.
-    pub(crate) fn label_extractor(&self) -> RouteLabelExtract {
-        use super::super::Route;
-        let Route {
-            parent_ref,
-            route_ref,
-            ..
-        } = &self.params;
-
-        RouteLabelExtract(parent_ref.clone(), route_ref.clone())
-    }
-}
```

Comment:
The main motivation for this change is so that we can use the `P` type to influence label extraction.